### ISSUE TITLE
Union parameters are lost in embedded struct

### ISSUE THREAD

**Issue by henrygab** [henrygab/010editor_binary_templates#17]
See example file at https://github.com/henrygab/010editor_binary_templates/blob/BugReports/Repository/BugReports/UnionParameterLost.bt

This repro's as of v9.0.1.

**Comment by henrygab** [comment]
This is not a bug.
The union has access to the parameter.  The embedded struct does not have access to the union's scope, and thus cannot see the variable.  This is correct scoping, working as it should.

Although not pretty code, it can be fixed.  See [example file update](https://github.com/henrygab/010editor_binary_templates/commit/4c9f67e8e6701ee02de7981461014521f8d7c74c#diff-f94ce2010e78d554ded330aa163afcbf) for an example.

Basically, the embedded struct must be declared with parameters, and the instantiation of the struct must pass the parameter, so that declarations within the inner struct can access the variable.
